components/script/canvas_state.rs

```diff
@@ -1892,6 +1951,19 @@ impl CanvasState {
     }
 }
 
+pub(crate) fn is_supported_canvas_size(size: Size2D<u64>) -> bool {
```

Comment:
Does this need to be `pub(crate)`? I would actually make this a static method on `DrawTargetSize` above.

```rust
impl DrawTargetSize
 ...
 fn is_supported_canvas_size()
...
```

Then you can call it using `Self::is_supported_canvas_size(...)`.

Reply:
Moved to DrawTarget struct. However may be better to move it to https://github.com/servo/servo/blob/main/components/shared/canvas/lib.rs  ??

Reply:
I would keep this in the `impl` block of `DrawTargetSize`.